src/linear_algebra/sesquilinear_form.lean

```diff
@@ -153,4 +193,95 @@ lemma ortho_comm {x y} : is_ortho B x y ↔ is_ortho B y x := H.is_refl.ortho_co
 
 end is_alt
 
+
+section orthogonal
+
+/-- The orthogonal complement of a submodule `N` with respect to some bilinear form is the set of
+elements `x` which are orthogonal to all elements of `N`; i.e., for all `y` in `N`, `B x y = 0`.
+
+Note that for general (neither symmetric nor antisymmetric) bilinear forms this definition has a
+chirality; in addition to this "left" orthogonal complement one could define a "right" orthogonal
+complement for which, for all `y` in `N`, `B y x = 0`.  This variant definition is not currently
+provided in mathlib. -/
+def orthogonal (B : M₁ →ₛₗ[I₁] M₁ →ₛₗ[I₂] R) (N : submodule R₁ M₁) : submodule R₁ M₁ :=
```

Comment:
I think it would make more sense to call this `submodule.orthogonal_bilin`, to bring it more in line with `submodule.orthogonal`. Also, I would inverse the order of the arguments, so that dot notation works with the submodule rather than with the bilinear form, again for the same reason.

Reply:
I changed that. In the end, `submodule.orthogonal` should be deleted in favor of this definition I think, but that will take some more work.